### ISSUE TITLE
page.getText(output='dict') return invalid bbox

### ISSUE THREAD

**Issue by gans** [rk700/PyMuPDF#180]
Extracting text information the bbox become always [0.0, 0.0, 0.0, 0.0]

`page.getText(output='dict')`

{'bbox': [0.0, 0.0, 0.0, 0.0],
             'lines': [{'bbox': (0.0, 0.0, 0.0, 0.0),
                        'dir': (1.0, 0.0),
                        'spans': [{'flags': 4,
                                   'font': 'ArialMT',
                                   'size': 7.99997615814209,
                                   'text': u'Editais'}],
                        'wmode': 0},

[test.pdf](https://github.com/rk700/PyMuPDF/files/2116253/test.pdf)

**Comment by JorjMcKie** [comment]
This is a weird one!
Same happens with `page.getText("html")`, so it isn't PyMuPDF.  And it does work for some blocks and lines. It also works for image blocks, but not always.
Same result also happens when trying `page.getTextBlocks()`. Using my other utility method `page.getTextWords()` delivers no empty bboxes, but words that are reported to cover the whole page more or less!
If you agree I would send your PDF to Artifex/MuPDF's bug reporting system ...

**Comment by JorjMcKie** [comment]
Update:

Initially, I thought I located the error: Mupdf used to have a bug that wouldn't correctly join character bboxes.
But that had been fixed before version 1.13.0.

I did my own bbox calculation to reconfirm the fix: Now all lines no longer have a bbox (0,0,0,0), **but still** the second and last bbox coordinates (corresponding to y0, y1 respectively) **are equal**, which makes the rectangle empty. The reason in these cases is that the contained characters all have a zero height, which makes **their** bboxes empty ... even though the line height is > 0. And **this** was the reason for what you observed. I cannot tell the true cause for this to happen ... maybe some error in handling the underlying font, or an error in the font program itself.

With this detail, I have submitted a bug report to MuPDF.

What I will do in the meantime nevertheless, is implementing a safeguard: if a character's bbox has zero height, I will increase the y1-coordinate by line height. This will yield correct "dict" output. Other places to apply this fix are methods `page.getTextBlocks()` and maybe `page.getTextWords()`.
The output formats corresponding to MuPDF "originals", "html" and "xml", will have to wait until a MuPDF fix (probably as late as November, v1.14).
